python/cuml/cuml/cluster/kmeans.pyx

```diff
@@ -31,15 +31,25 @@ from cuml.metrics.distance_type cimport DistanceType
 
 cdef _kmeans_init_params(kmeans, lib.KMeansParams& params):
     """Initialize a passed KMeansParams instance from a KMeans instance."""
+    cdef bool multi_gpu = kmeans._multi_gpu
+
     params.n_clusters = kmeans.n_clusters
     params.max_iter = kmeans.max_iter
     params.tol = kmeans.tol
-    params.rng_state.seed = check_random_seed(kmeans.random_state)
     params.verbosity = <level_enum>(<int>kmeans.verbose)
     params.metric = DistanceType.L2Expanded
     params.batch_samples = int(kmeans.max_samples_per_batch)
     params.oversampling_factor = kmeans.oversampling_factor
 
+    # Ensure random_state is set when running on multi-gpu
+    if multi_gpu and kmeans.random_state is None:
```

Comment:
This is more of a philosophical question, but you think it's better to err here rather than to set a seed for the user?

Reply:
We have to error here, if each partition uses a different seed you get incorrect results. Erroring here is to catch cases where the consumer of `KMeansMG` was passing `random_state=None` and inform the user of their mistake.

In `cuml.dask.cluster.KMeans` we handle this case transparently, generating a single seed on the client before calling on every partition. For users like nemo-curator that are using the `KMeansMG` class directly we want to error so they don't let a bug sneak through.